hubblestack/extmods/hubble_mods/win_auditpol.py

```diff
@@ -0,0 +1,158 @@
+        return False, 'This audit module only runs on windows'
+    return True
+
+
+def validate_params(block_id, block_dict, extra_args=None):
+    """
+    Validate all mandatory params required for this module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': 'Distribution Group Management', 'status': True},
+                  'caller': 'Audit'}
+
+    Raises:
+        HubbleCheckValidationError: For any validation error
+    """
+    log.debug('Module: win_auditpol. Start validating params for check-id: {0}'.format(block_id))
+    error = {}
+    chain_args = extra_args.get('chaining_args')
+    chained_name = runner_utils.get_chained_param(chain_args)
+    name = runner_utils.get_param_for_module(block_id, block_dict, 'name')
+    # fetch required param
+    if not name and not chained_name:
+        error['name'] = 'Mandatory parameter: name not found for id: %s' % block_id
+
+    if error:
+        raise HubbleCheckValidationError(error)
+
+    log.debug('Validation success for check-id: {0}'.format(block_id))
+
+
+def execute(block_id, block_dict, extra_args=None):
+    """
+    Execute the module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': 'Distribution Group Management', 'status': True},
+                  'caller': 'Audit'}
+
+    returns:
+        tuple of result(value) and status(boolean)
+    """
+    log.debug('Executing win_auditpol module for id: {0}'.format(block_id))
+    chain_args = extra_args.get('chaining_args')
+    # fetch required param
+    name = runner_utils.get_chained_param(chain_args)
+    if not name:
+        name = runner_utils.get_param_for_module(block_id, block_dict, 'name')
+
+    __auditdata__ = _auditpol_import()
+    if name in __auditdata__:
+        audit_value = __auditdata__[name]
+        return runner_utils.prepare_positive_result_for_module(block_id, {name: audit_value})
+
+    return runner_utils.prepare_negative_result_for_module(block_id, 'policy_not_found')
+
+
+def get_filtered_params_to_log(block_id, block_dict, extra_args=None):
+    """
+    For getting params to log, in non-verbose logging
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': 'Distribution Group Management', 'status': True},
+                  'caller': 'Audit'}
+    """
+    log.debug('Module: win_auditpol get_filtered_params_to_log for id: {0}'.format(block_id))
+    chain_args = extra_args.get('chaining_args')
+    # fetch required param
+    name = runner_utils.get_chained_param(chain_args)
```

Comment:
I have made some changes in the `get_chained_param` function. This code can later be modified accordingly. You won't need to write line 131, you can simply write `runner_utils.get_chained_param(extra_args)`

Reply:
fixed